### ISSUE TITLE
rustc-1.54.0-src.patch fails to apply on macOS 13

### ISSUE THREAD

**Issue by ryandesign** [thepowersgang/mrustc#298]
On macOS 13 I'm unable to apply your patch rustc-1.54.0-src.patch to the rust 1.54.0 sources:

```
patching file 'compiler/rustc_ast/src/ast.rs'
patching file 'compiler/rustc_hir/src/hir.rs'
patching file 'compiler/rustc_middle/src/mir/interpret/error.rs'
1 out of 1 hunks failed--saving rejects to 'compiler/rustc_middle/src/mir/interpret/error.rs.rej'
patching file 'compiler/rustc_middle/src/mir/mod.rs'
1 out of 1 hunks failed--saving rejects to 'compiler/rustc_middle/src/mir/mod.rs.rej'
patching file 'compiler/rustc_middle/src/thir.rs'
1 out of 1 hunks failed--saving rejects to 'compiler/rustc_middle/src/thir.rs.rej'
patching file 'compiler/rustc_mir/src/interpret/place.rs'
2 out of 2 hunks failed--saving rejects to 'compiler/rustc_mir/src/interpret/place.rs.rej'
patching file 'compiler/rustc_mir/src/interpret/operand.rs'
2 out of 2 hunks failed--saving rejects to 'compiler/rustc_mir/src/interpret/operand.rs.rej'
patching file 'vendor/crc32fast/src/specialized/mod.rs'
patching file 'library/stdarch/crates/std_detect/src/detect/mod.rs'
1 out of 1 hunks failed--saving rejects to 'library/stdarch/crates/std_detect/src/detect/mod.rs.rej'
patching file 'vendor/ppv-lite86/src/lib.rs'
patching file 'vendor/curl/src/lib.rs'
```

On macOS 12 it works:

```
patching file compiler/rustc_ast/src/ast.rs
patching file compiler/rustc_hir/src/hir.rs
patching file compiler/rustc_middle/src/mir/interpret/error.rs
Hunk #1 succeeded at 452 with fuzz 1.
patching file compiler/rustc_middle/src/mir/mod.rs
Hunk #1 succeeded at 2203 with fuzz 1.
patching file compiler/rustc_middle/src/thir.rs
Hunk #1 succeeded at 147 with fuzz 1.
patching file compiler/rustc_mir/src/interpret/place.rs
Hunk #1 succeeded at 91 with fuzz 1.
Hunk #2 succeeded at 100 with fuzz 1.
patching file compiler/rustc_mir/src/interpret/operand.rs
Hunk #1 succeeded at 35 with fuzz 1.
Hunk #2 succeeded at 90 with fuzz 1.
patching file vendor/crc32fast/src/specialized/mod.rs
patching file library/stdarch/crates/std_detect/src/detect/mod.rs
Hunk #1 succeeded at 88 with fuzz 1.
patching file vendor/ppv-lite86/src/lib.rs
Hunk #1 succeeded at 12 with fuzz 1.
patching file vendor/curl/src/lib.rs
```

macOS 13 includes a different version of the `patch` program that I've found is much pickier than the `patch` program in macOS 12 and earlier. The new one doesn't seem to like applying patches that require fuzz.

Could you regenerate your patchfile(s) so that they apply without fuzz?

**Comment by ryandesign** [comment]
On macOS 13:

```
% /usr/bin/patch --version
patch 2.0-12u11-Apple
```

On macOS 12:

```
% /usr/bin/patch --version
patch 2.5.8
Copyright (C) 1988 Larry Wall
Copyright (C) 2002 Free Software Foundation, Inc.

This program comes with NO WARRANTY, to the extent permitted by law.
You may redistribute copies of this program
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING.

written by Larry Wall and Paul Eggert
```

**Comment by ryandesign** [comment]
> Could you regenerate your patchfile(s) so that they apply without fuzz?

Like this for example: https://github.com/macports/macports-ports/blob/9d495c5f9e04ff36646fecb26b0417be54e2567c/lang/mrustc/files/rustc-1.54.0-src.patch.patch

**Comment by thepowersgang** [comment]
@ryandesign Can you confirm that the issue was fixed by #310
